### ISSUE TITLE
Doc & Examples - make it explicit how to use lib modules from bin, test.

### ISSUE THREAD

**Issue by LAC-Tech** [ocaml/dune#6551]
Ie, from the Quickstart guide (emphasis mine):

"The bin directory holds a skeleton for the executable program. Within the modules in this directory, **you can access the modules in your lib under the namespace project_name.Mod, where project_name is replaced with the name of your project and Mod corresponds to the name of the file in the lib directory**. You can run the executable with dune exec project_name. See Building a Hello World Program From Scratch for an example of specifying an executable manually and executable for details."

So stuff I tried:

```ocaml
open project_name.Mod
```
^ "Syntax error". Oh that's right, ocaml modules are capitalized. Ok...

```ocaml
open Project_Name.mod
```
^ "Unbound module.". Maybe it needs to be in the libraries list of the dune file? No, it's already there. Maybe I should look at the examples? No, they don't use modules from lib inside bin.

Embarrassingly I spent maybe a couple of hours trying to figure this out, and was only saved by one Geoff from the dune channel in the discord, who told me that it was `Project_name`.

# Objections

## Isn't it obvious how dune does this, if you know ocaml?

I would imagine you get a lot of new people trying out dune.

Here's a scenario:

- programmer hears about ocaml on popular programming discussion forum where it's often praised.
- programmers googles a bit, finds everyone recommends dune
- programmer reads dune quick start guide
- programmer has zero intuitive understanding of the capitalisation (or not) of ocaml modules, can't get basic code running, gives up.

**Comment by nojb** [comment]
I agree that Dune is often the first port of entry for beginners and that a beginner-friendly tutorial/quickstart would be a good thing.

**Comment by davesnx** [comment]
Totally @nojb 

Talking in Discord, maybe there's another possible fix on ocaml/ocaml related with `Unbound module`. The compiler has some sort of algorithm to check for similarities (Levenshtein distance, I believe) or typos on values, maybe could we run the same for modules or that might require dune info (?).

**Comment by Alizter** [comment]
cc @emillon what should we do about this?
